### PR TITLE
Fixes for delayed start of docker daemon

### PR DESCRIPTION
**Pull request by amitlimaye**
#### Description
*Changes proposed in this pull request.*

#### Test plan
*Outline the test plan used to test this change before merging it.*

> Fixes #.

### CODE REVIEW

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "delayeddocker",
    "component": "enforcerd",
    "pr-id": "1213",
    "commit-sha": "6cd541ed3952b3414109c7014b636be0c109b89b"
  },
  {
    "project": "delayeddocker",
    "component": "trireme-lib",
    "pr-id": "778",
    "commit-sha": "abf6334c5c8fb6d3998c558c32f1820d18b8ffad"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "delayeddocker",
    "component": "trireme-lib",
    "pr-id": "778",
    "commit-sha": "72151ac896fadf24e5c60edea49146ada7669c6e"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "delayeddocker",
    "component": "trireme-lib",
    "pr-id": "778",
    "commit-sha": "72151ac896fadf24e5c60edea49146ada7669c6e"
  },
  {
    "project": "delayeddocker",
    "component": "enforcerd",
    "pr-id": "1220",
    "commit-sha": "6cd541ed3952b3414109c7014b636be0c109b89b"
  }
]
```

File: monitor/internal/docker/monitor.go [hunk 1]
```diff
@@ -654,14 +657,27 @@ func (d *DockerMonitor) setupDockerDaemon() (err error) {
...
-		for errg := d.setupDockerDaemon(); errg != nil; {
+	go func(gctx context.Context) {
+
+		for {
+			errg := d.setupDockerDaemon()
+			if errg == nil {
+				if err := d.initMonitor(gctx); err != nil {
+					zap.L().Debug("Unable to init monitor")
+				}
+				break
```
Comment (dstiliadis) [inline monitor/internal/docker/monitor.go:672]: I think this needs to be an error, so users can see that even though the docker daemon is there, we were never able to start it

File: monitor/internal/docker/monitor.go [hunk 2]
```diff
@@ -654,14 +657,27 @@ func (d *DockerMonitor) setupDockerDaemon() (err error) {
 func (d *DockerMonitor) waitForDockerDaemon(ctx context.Context) (err error) {
 
 	done := make(chan bool)
-	go func() {
-		for errg := d.setupDockerDaemon(); errg != nil; {
+	go func(gctx context.Context) {
```
Comment (dstiliadis) [inline monitor/internal/docker/monitor.go:664]: For user experience, let's add an Info message here: "Trying to initialize docker monitor ... "

File: monitor/internal/docker/monitor.go (hunk 1 above)
Comment (dstiliadis) [inline monitor/internal/docker/monitor.go:672]: When we finally succeed, again for user experience, let's put an Info message "Docker monitor initialized"

With these messages, we will know that we are trying .. and whether we succeeded or eventually failed" 
We don't need messages for retry. 

Makes sense?

**Review by dstiliadis — CHANGES_REQUESTED** [review]
See inline.

File: monitor/internal/docker/monitor.go (hunk 2 above)
Comment (amitlimaye) [reply monitor/internal/docker/monitor.go:664]: Done

File: monitor/internal/docker/monitor.go (hunk 1 above)
Comment (amitlimaye) [reply monitor/internal/docker/monitor.go:672]: Added a message here as well when we suceed

Comment (amitlimaye) [reply monitor/internal/docker/monitor.go:672]: Done

**Review by dstiliadis — APPROVED** [review]

**Pull request merged**